### PR TITLE
docs: add js docs to the query method and StorefrontResponse type

### PR DESCRIPTION
**Pull request by mdarrik**
## Why are these changes introduced?

Addresses [ENG-8409](https://nacelle.atlassian.net/browse/ENG-8409) - improves documentation of the v2 sdk's query method 

## What is this pull request doing?

1. Adds js-doc comments to the sdk's query method.
2. Adds js-doc comments to the sdk's StorefrontResponse type.

## How to Test

1. Use `npm run build` && `npm link` to import the sdk into another project. (Can also use the same file in this project)
2. Instantiate an sdk client
3. Review the in-editor information for the `client.query` method & the different fields returned in the response.  

[ENG-8409]: https://nacelle.atlassian.net/browse/ENG-8409?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/storefront-sdk/src/client/index.ts [hunk 1]
```diff
@@ -218,6 +227,13 @@ export class StorefrontClient {
 		return result;
 	}
 
+	/**
+	 * @param {QueryParams} queryParams - The arguments for your graphql query. the query field accepts strings, GraphQL AST's, or TypedDocumentNodes. If using typescript, you can use {@link https://github.com/dotansimha/graphql-typed-document-node#how-to-use TypedDocumentNode} to get typed responses on your queries.
```
Comment (NWRichmond) [inline packages/storefront-sdk/src/client/index.ts:231]:
```suggestion
	 * @param {QueryParams} queryParams - The arguments for your GraphQL query. The `query` field accepts strings, GraphQL ASTs, or TypedDocumentNodes. If using TypeScript, you can use {@link https://github.com/dotansimha/graphql-typed-document-node#how-to-use TypedDocumentNode} to get typed responses on your queries.
```

File: packages/storefront-sdk/src/client/index.ts [hunk 2]
```diff
@@ -218,6 +227,13 @@ export class StorefrontClient {
...
+	 * @returns {Promise<StorefrontResponse>} storefrontResponse - the results of your query. If the query succeeds, the data will be on the `data` field. If there are errors, they will be contained within the `error` field.
+	 *
+	 * @example
+	 * const {data, error} = await client.query({query: myQuery, variables: queryVariables});
```
Comment (NWRichmond) [inline packages/storefront-sdk/src/client/index.ts:235]:
```suggestion
	 * const { data, error } = await client.query({
	 * 	query: myQuery,
	 * 	variables: queryVariables
	 * });
```

**Review by NWRichmond — APPROVED** [review]
Cool! I'm getting good feedback in VSCode. Nice use of `@link`.

I added some nitpicky suggestions; they don't merit a re-review.

**Review by KrisQ — APPROVED** [review]
Looks good!

**Pull request merged**